### DISCUSSION TITLE
WinForms commitment for the future

### DISCUSSION THREAD

**Discussion by AllenAtTitanium** [dotnet/winforms#8506] [Q&A] [answered]
This is just something I'm curious about for future development, because I may need to start one or several new projects in the future.

I know that WinForms is in .Net 6+, but what isn't done, what's broken, what controls don't work right yet, or haven't been implemented?

Has WinForms on .NET hit feature parity with .NET Framework?

Will new features come to the WinForms on .NET or is it in Maintenance mode? I've seen blog posts say this, but I've not seen a single one cite a source.

Would Microsoft ever obsolete WinForms if a shiny new stack shifts the development world and actually succeeds where Maui, Win UI, UWP and so on have failed?

I'm looking into where the development community is going and it looks like WinForms is in fact not "dead", as many seem to be saying. B2B software often does "okay" with what I'm seeing from WinForms, but I need some form of certainty, as there's no current clear indicators.

I feel that WinForms was ported to .NET 6+ because there was backlash, from an adoption standpoint, enterprise software (and ISVs) were probably pushing back against it. A new framework is great, but WinForms has, hands down, the most productive WYSIWYG editor out there for software development. When you need to get it done, WinForms just works.

**Comment by KlausLoeffelmann** [comment, marked as answer]
> Will new features come to the WinForms on .NET or is it in Maintenance mode? I've seen blog posts say this, but I've not seen a single one cite a source.

https://devblogs.microsoft.com/dotnet/category/winforms/

I think this will answer most of your questions! :-)

Thanks,

Klaus
